test/lib/Controller.test.ts

```diff
@@ -15,30 +15,32 @@ describe('Controller.ts tests', () => {
 
 	it('getAllEngines returns enabled/disabled engines', async() => {
 		const engines: RuleEngine[] = await Controller.getAllEngines();
-		const names: string[] = engines.map(e => e.getName());
-
-		expect(engines.length, names + '').to.equal(9);
-		expect(names).to.contain(ENGINE.ESLINT);
-		expect(names).to.contain(ENGINE.ESLINT_LWC);
-		expect(names).to.contain(ENGINE.ESLINT_TYPESCRIPT);
-		expect(names).to.contain(ENGINE.ESLINT_CUSTOM);
-		expect(names).to.contain(ENGINE.PMD);
-		expect(names).to.contain(ENGINE.PMD_CUSTOM);
-		expect(names).to.contain(ENGINE.RETIRE_JS);
-		expect(names).to.contain(ENGINE.CPD);
-		expect(names).to.contain(ENGINE.SFGE);
+		const names: string[] = engines.map(e => e.constructor.name);
```

Comment:
The DFA and Pathless SFGE variants share the same `.getName()` result, so we have to check the constructors instead to determine that the expected engines were returned.